src/components/TimeCardPage/CellTypes/CommentCell.tsx

```diff
@@ -1,43 +1,219 @@
-        if (props.comments === undefined) {
-            //Create one empty comment 
-            setComments([{
-                AuthorID:"<TODO Fill this in at some point>", 
-                Type: CommentType.Comment, 
-                Timestamp: moment().tz(TIMEZONE).unix(), 
-                Content:"" 
-            }]); 
-        } 
-    }, []);  
-    
-
-    const onUpdate = (event) => {
-        comments[0].Content = event.target.value; 
-        props.setComment("Comment", comments); 
-    } 
-
-    if (comments !== undefined && comments.length > 0) {
-        return <input defaultValue={comments[0].Content} onChange={onUpdate} />
-    } else {
-        return <input placeholder="Any comments?" onChange={onUpdate} />
+  comments: CommentSchema[] | undefined;
+  setComment: Function; // TODO: fix type
+}
+
+// setComments can be used for comments or reports
+interface ShowCommentModalProps {
+  comments: CommentSchema[];
+  currentComment: CommentSchema;
+  setComments: Function;
+  icon; // TODO: add type
+  color: Color;
+  isEditable: boolean;
+}
+
+function ShowCommentModal({
+  comments,
+  currentComment,
+  setComments,
+  icon,
+  color,
+  isEditable,
+}: ShowCommentModalProps) {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [displayedComment, setDisplayedComment] = useState(
+    currentComment
+  );
+  const user = useContext(UserContext);
+
+  const EditableControls = () => {
+    const {
+      isEditing,
+      getSubmitButtonProps,
+      getCancelButtonProps,
+      getEditButtonProps,
+    } = useEditableControls();
+
+    // TODO: change this later to reflect figma
+    return isEditing ? (
+      <ButtonGroup justifyContent="center" size="sm">
+        <Button leftIcon={<CheckIcon />} {...getSubmitButtonProps()} />
+        <Button leftIcon={<CloseIcon />} {...getCancelButtonProps()} />
+      </ButtonGroup>
+    ) : (
+      <Flex justifyContent="right">
+        <Button size="sm" leftIcon={<EditIcon />} {...getEditButtonProps()} />
+      </Flex>
+    );
+  };
+
+  const elevatedUserPrivileges =
+    user?.Type === "Supervisor" || user?.Type === "Admin";
+
+  // TODO: probably need to add some more validation
+  const addNewComment = (value: string) => {
+    if (value !== "") {
+      setDisplayedComment(createNewComment(user, displayedComment.Type, value));
     }
+  };
+
+  const saveComment = () => {
+    // Do we want people to be able to delete comments by saving empty comments?
+    currentComment.State = CellStatus.Deleted
+    setComments(getAllActiveCommentsOfType(displayedComment.Type, [...comments, displayedComment]));
+    // TODO: save to DB
+  };
+
+  const deleteComment = () => {
+    currentComment.State = CellStatus.Deleted
+    setComments(getAllActiveCommentsOfType(currentComment.Type, comments));
+    // TODO: save to DB
+  }
+
+  // TODO: make the editable work as intended later, without the odd preview box and whatever
```

Comment:
Not sure if this has been done yet but just so that it doesnt get lost making a ticket describing this would be really helpful!